File: typescript-client-sdk/src/sdk/models/shared/projectrunsresponsepayload.ts
import { SpeakeasyBase, SpeakeasyMetadata } from "../../../internal/utils";
import { ProjectStatusResponsePayload } from "./projectstatusresponsepayload";
import { Expose, Type } from "class-transformer";


export class ProjectRunsResponsePayload extends SpeakeasyBase {
  @SpeakeasyMetadata()
  @Expose({ name: "nextPage" })
  nextPage: Record<string, any>;

  @SpeakeasyMetadata()
  @Expose({ name: "previousPage" })
  previousPage: Record<string, any>;

  @SpeakeasyMetadata({ elemType: ProjectStatusResponsePayload })
  @Expose({ name: "runs" })
  @Type(() => ProjectStatusResponsePayload)
  runs: ProjectStatusResponsePayload[];

  @SpeakeasyMetadata()
  @Expose({ name: "traceId" })
  traceId: string;
}

File: go-client-sdk/pkg/models/shared/projectrunsresponsepayload.go
package shared

type ProjectRunsResponsePayload struct {
	NextPage     map[string]interface{}         `json:"nextPage"`
	PreviousPage map[string]interface{}         `json:"previousPage"`
	Runs         []ProjectStatusResponsePayload `json:"runs"`
	TraceID      string                         `json:"traceId"`
}
